storage_backblaze.go

```diff
@@ -0,0 +1,193 @@
+/*
+ * knoxite
+ *     Copyright (c) 2016, Christian Muehlhaeuser <muesli@gmail.com>
+ *     Copyright (c) 2016, Nicolas Martin <penguwingithub@gmail.com>
+ *
+ *   For license see LICENSE.txt
+ */
+
+package knoxite
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"gopkg.in/kothar/go-backblaze.v0"
+)
+
+// StorageBackblaze stores data on a remote Backblaze
+type StorageBackblaze struct {
+	url            url.URL
+	chunkFile      string
+	snapshotFile   string
+	repositoryFile string
+	// bucketPrefix   string
+	bucket    *backblaze.Bucket
+	backblaze *backblaze.B2
+}
+
+// NewStorageBackblaze returns a
+func NewStorageBackblaze(URL url.URL) (*StorageBackblaze, error) {
+	// Checking username and password
+	if URL.User.Username() == "" {
+		return &StorageBackblaze{}, nil
+	}
+	pw, pwexist := URL.User.Password()
+	if !pwexist {
+		return &StorageBackblaze{}, ErrInvalidPassword
+	}
+
+	// Creating a new Client for accessing the B2 API
+	cl, err := backblaze.NewB2(backblaze.Credentials{
+		AccountID:      URL.User.Username(),
+		ApplicationKey: pw,
+	})
+	if err != nil {
+		return &StorageBackblaze{}, err
+	}
+
+	// Creating the bucket prefixes
+	bucketPrefix := strings.Split(URL.Path, "/")
+	if len(bucketPrefix) != 2 {
+		return &StorageBackblaze{}, ErrInvalidRepositoryURL
+	}
+
+	// TODO: Make me handsome
+	var bucket *backblaze.Bucket
+	bucket, err = cl.Bucket(bucketPrefix[1])
+	if err != nil || bucket == nil {
+		// Bucket probably doesn't exists
+		bucket, err = cl.CreateBucket(bucketPrefix[1], backblaze.AllPrivate)
+		if err != nil {
+			// Bucket exists but we don't have access on it
+			return &StorageBackblaze{}, err
+		}
+		if bucket == nil {
+			panic("bucket nil")
+		}
+		return &StorageBackblaze{
+			url:            URL,
+			chunkFile:      bucketPrefix[1] + "-chunks",
+			snapshotFile:   bucketPrefix[1] + "-snapshots",
+			repositoryFile: bucketPrefix[1] + "-repository",
+			// bucketPrefix:   bucketPrefix[1],
+			bucket:    bucket,
+			backblaze: cl,
+		}, nil
+	}
+	return &StorageBackblaze{
```

Comment:
You can avoid having the same return block twice here by removing the one above and not returning within the if-clause.